PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/explorer/ConnectorToolWindow.java

```diff
@@ -0,0 +1,135 @@
+import com.microsoft.azure.toolkit.intellij.connector.ConnectorDialog;
+import com.microsoft.azure.toolkit.intellij.connector.ModuleResource;
+import com.microsoft.azure.toolkit.intellij.connector.Resource;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import java.util.Objects;
+import java.util.Optional;
+
+public class ConnectorToolWindow extends SimpleToolWindowPanel {
+    private final Tree tree;
+
+    private final Project project;
+
+    public ConnectorToolWindow(@NotNull final Project project) {
+        super(true);
+        this.project = project;
+        this.tree = new ResourceConnectorTree(project);
+        new TreeSpeedSearch(tree);
+        ActionToolbarImpl actionToolbar = this.initToolbar();
+        actionToolbar.setTargetComponent(this.tree);
+        actionToolbar.setForceMinimumSize(true);
+        this.setContent(this.tree);
+        this.setToolbar(actionToolbar);
+    }
+
+    private ActionToolbarImpl initToolbar() {
+        final DefaultActionGroup group = new DefaultActionGroup();
+        final CommonActionsManager manager = CommonActionsManager.getInstance();
+        group.add(new RefreshAction());
+        group.add(new AddAction());
+        group.add(new RemoveAction());
+        group.addSeparator();
+        // expand and collapse
+        final DefaultTreeExpander expander = new DefaultTreeExpander(this.tree);
+        group.add(manager.createExpandAllAction(expander, this.tree));
+        group.add(manager.createCollapseAllAction(expander, this.tree));
+        return new ActionToolbarImpl(ActionPlaces.TOOLBAR, group, true);
+    }
+
+    private class RefreshAction extends com.intellij.ide.actions.RefreshAction {
+        private boolean loading = false;
+
+        RefreshAction() {
+            super(IdeBundle.messagePointer("action.refresh"), IdeBundle.messagePointer("action.refresh"), AllIcons.Actions.Refresh);
+        }
+
+        @Override
+        public final void actionPerformed(@NotNull final AnActionEvent e) {
+            this.loading = true;
+            ((AzureTree) ConnectorToolWindow.this.tree).loadNodes();
+            this.loading = false;
+        }
+
+        @Override
+        public final void update(@NotNull final AnActionEvent event) {
+            final Presentation presentation = event.getPresentation();
+            final Icon icon = loading ? new AnimatedIcon.Default() : this.getTemplatePresentation().getIcon();
+            presentation.setIcon(icon);
+            presentation.setEnabled(!loading);
+        }
+    }
+
+    private class AddAction extends AnAction {
+
+        AddAction() {
+            super("Add", "Add new Azure resource connector", AllIcons.General.Add);
+        }
+
+        @Override
+        public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
+            final ConnectorDialog<? extends Resource, ModuleResource> dialog = new ConnectorDialog<>(project);
+            dialog.show();
```

Comment:
should set the event source as the default resource/consumer in the connector dialog

Reply:
updated to set